subworkflows/local/assemble/main.nf

```diff
@@ -0,0 +1,222 @@
+include { FLYE } from '../../../modules/nf-core/flye/main'
+include { HIFIASM } from '../../../modules/nf-core/hifiasm/main'
+include { HIFIASM as HIFIASM_ONT } from '../../../modules/nf-core/hifiasm/main'
+include { GFA_2_FA } from '../../../modules/local/gfa2fa/main'
+include { MAP_TO_ASSEMBLY } from '../mapping/map_to_assembly/main'
+include { MAP_TO_REF } from '../mapping/map_to_ref/main'
+include { RUN_QUAST } from '../qc/quast/main'
+include { RUN_BUSCO } from '../qc/busco/main'
+include { MERQURY_QC } from '../qc/merqury/main'
+include { RUN_LIFTOFF } from '../liftoff/main'
+include { RAGTAG_SCAFFOLD } from '../../../modules/local/ragtag/main'
+
+
+workflow ASSEMBLE {
+  take:
+  ont_reads // meta, reads
+  hifi_reads // meta, reads
+  ch_input
+  genomescope_out
+  meryl_kmers
+
+  main:
+  // Empty channels 
+  Channel.empty().set { ch_refs }
+  Channel.empty().set { ch_ref_bam }
+  Channel.empty().set { ch_assembly_bam }
+  Channel.empty().set { ch_assembly }
+  Channel.empty().set { assembly_quast_reports }
+  Channel.empty().set { assembly_busco_reports }
+  Channel.empty().set { assembly_merqury_reports }
+  Channel.empty().set { flye_inputs }
+  Channel.empty().set { hifiasm_inputs }
+  Channel.empty().set { longreads }
+
+  if (params.use_ref) {
+    ch_input
+      .map { row -> [row.meta, row.ref_fasta] }
+      .set { ch_refs }
+  }
+
+  if (params.skip_assembly) {
+    // Sample sheet layout when skipping assembly
+    // sample,ontreads,assembly,ref_fasta,ref_gff
+    ch_input
+      .map { row -> [row.meta, row.assembly] }
+      .set { ch_assembly }
+  }
+  if (!params.skip_assembly) {
+    def hifi_only = params.hifi && !params.ont ? true : false
+    // Define inputs for flye
+    if (params.assembler == "flye") {
+      if (params.hifi) {
+        if (!hifi_only) {
```

Comment:
This could be also checked at the `PIPELINE_INITIALISATION` step. But as there is no process ran before hand there might not be any need. The aim is to throw an error the quickest as possible if the user does not specify the correct set of argument.